cmd/stolonctl/upgrade.go

```diff
@@ -33,13 +36,38 @@ var (
 
 func init() {
 	stolonctlCmd.AddCommand(upgradeCmd)
+	upgradeCmd.Flags().StringVar(&clusterConfig.PostgresHost, "postgres-host",
+		defaults.PostgresHost, "Hostname for connection to old stolon PostreSQL host")
+	upgradeCmd.Flags().StringVar(&clusterConfig.PostgresPort, "postgres-port",
+		defaults.PostgresPort, "Port for connection to old stolon PostgreSQL host")
+	upgradeCmd.Flags().StringVar(&clusterConfig.PostgresUser, "postgres-user",
+		defaults.PostgresUser, "Username for connection to old stolon PostgreSQL host")
+	upgradeCmd.Flags().StringVar(&clusterConfig.PostgresPassword, "postgres-password",
+		"", "Password for connection to ols stolon PostgreSQL host")
+	upgradeCmd.Flags().StringVar(&clusterConfig.PostgresBackupPath,
+		"postgres-backup-path", defaults.PostgresBackupPath,
+		"Path to store backup of old stolon PostgreSQL data")
+
+	for env, flag := range envs {
+		cmdFlag := upgradeCmd.Flags().Lookup(flag)
+		if value := os.Getenv(env); value != "" {
+			if cmdFlag != nil {
+				cmdFlag.Value.Set(value)
+			}
+		}
+	}
+
 }
 
 func upgrade(ccmd *cobra.Command, args []string) error {
 	if err := clusterConfig.CheckConfig(); err != nil {
 		return trace.Wrap(err)
 	}
 
+	if err := clusterConfig.CheckPostgresParams(); err != nil {
```

Comment:
This could be done as part of `clusterConfig.CheckConfig`:
```go
func (r clusterConfig) Check() error {
	// check cluster attributes
	if err := r.postgresConfig.check(); err != nil {
		return trace.Wrap(err)
	}
}
```